@AH-64D Apache Longbow/Addons/fza_ah64_sfmplus/functions/fn_getInput.sqf

```diff
@@ -20,38 +20,73 @@ params ["_heli", "_deltaTime"];
+private _configVehicles	   = configFile >> "CfgVehicles" >> typeof _heli;
+private _flightModel 	   = getText (_configVehicles>> "flightModel");
 private _pitchTorque       = getNumber (_config >> "cyclicPitchTorque");
 private _rollTorque        = getNumber (_config >> "cyclicRollTorque");
 private _yawTorque         = getNumber (_config >> "pedalYawTorque");
 
+private _hydFailure        = false;
 private _tailRtrFixed      = false;
 
 private _priHydPumpDamage  = _heli getHitPointDamage "hit_hyd_pripump";
 private _priHydPSI         = _heli getVariable "fza_systems_priHydPsi";
 
 private _utilHydPumpDamage = _heli getHitPointDamage "hit_hyd_utilpump";
+private _utilHydPSI        = _heli getVariable "fza_systems_utilHydPsi";
 private _utilLevel_pct     = _heli getVariable "fza_systems_utilLevel_pct";
 
 private _accOn 		   	   = _heli getVariable "fza_systems_accOn";
 
-private _collectiveVal     = _heli animationSourcePhase "collective";
 private _cyclicFwdAft      = _heli animationSourcePhase "cyclicForward";
 private _cyclicLeftRight   = _heli animationSourcePhase "cyclicAside";
 private _pedalLeftRight    = (inputAction "HeliRudderRight") - (inputAction "HeliRudderLeft");
 
 private _tailRtrDamage     = _heli getHitPointDamage "hitvrotor";
 
 private _collectiveOut     = 0.0;
-if (fza_ah64_sfmPlusKeyboardOnly) then {
-	_collectiveVal = [_collectiveVal, 0.5, 1.0] call BIS_fnc_clamp;
-	_collectiveOut = linearConversion[ 0.5, 1.0, _collectiveVal, 0.0, 2.0];
+if (_flightModel == "SFMPlus") then {
+	private _collectiveVal = _heli animationSourcePhase "collective";
 
-	private _V_mps            = abs vectorMagnitude [velocity _heli select 0, velocity _heli select 1];
-	private _collectiveOutMod = linearConversion [0.00, 12.35, _V_mps, 0.775, 0.853, true];
+	if (fza_ah64_sfmPlusKeyboardOnly) then {
+		_collectiveVal 		   = [_collectiveVal, 0.5, 1.0] call BIS_fnc_clamp;
+		_collectiveOut 	       = linearConversion[ 0.5, 1.0, _collectiveVal, 0.0, 2.0];
 
-	_collectiveOut = [_collectiveOut, 0.0, _collectiveOutMod] call BIS_fnc_clamp;
+		private _V_mps            = vectorMagnitude [velocityModelSpace _heli # 0, velocityModelSpace _heli # 1];
+		private _collectiveOutMod = linearConversion [0.00, 12.35, _V_mps, 0.775, 0.853, true];
+
+		_collectiveOut = [_collectiveOut, 0.0, _collectiveOutMod] call BIS_fnc_clamp;
+	} else {
+		_collectiveOut = linearConversion[-1.0, 1.0, _collectiveVal, 0.0, 1.0];
+	};
 } else {
-	_collectiveOut = linearConversion[-1.0, 1.0, _collectiveVal, 0.0, 1.0];
+	//systemChat format ["HeliSim Input Handler"];
+	//Keyboard collective
+	private _keyCollectiveUp = inputAction "HeliCollectiveRaise";
+	private _keyCollectiveDn = inputAction "HeliCollectiveLower";
+	//Joystick collective
+	private _joyCollectiveUp = inputAction "HeliCollectiveRaiseCont";
+	private _joyCollectiveDn = inputAction "HeliCollectiveLowerCont";
+	private _collectiveVal   = _heli getVariable "fza_sfmplus_collectiveVal";
```

Comment:
What is the difference between `fza_sfmplus_collectiveVal` and `fza_sfmplus_collectiveOutput`?

Reply:
Collective val is the raw animation source phase for SFM and then it's the stored collective position for HeliSim. It's what allows the keyboard collective to function.

Reply:
Think of it as collectiveVal is unprocessed, fza_sfmplus_collectiveOuput is processed.